arch/arm/src/armv7-m/gnu/arm_exception.S

```diff
@@ -321,9 +322,9 @@ exception_common:
 	.bss
 	.global	g_intstackalloc
 	.global	g_intstackbase
-	.align	8
+	.balign	8
 g_intstackalloc:
-	.skip	((CONFIG_ARCH_INTERRUPTSTACK + 4) & ~7)
+	.skip	(CONFIG_ARCH_INTERRUPTSTACK & ~7)
```

Comment:
Shouldn't we align up here? `((CONFIG_ARCH_INTERRUPTSTACK + 7) & ~7)`

Reply:
@Ouss4 
I agree.

Reply:
the alignment need sync with the code in stack setup and dump. for example here is line 70:
```
#  if defined(CONFIG_BUILD_PROTECTED) && CONFIG_ARCH_INTERRUPTSTACK < 8
```
if we select the align up, the above logic is wrong and should change to CONFIG_ARCH_INTERRUPTSTACK < 1.
So, I would like to keep the logic as simple as possible instead bringing up more inconsistentence like this. The round down logic is more suitable choice because I don't think there is much difference betwen 2000 and 2008 if user set CONFIG_ARCH_INTERRUPTSTACK to 2006. The consistent round strategy across all related code is more important.

Reply:
@xiaoxiang781216 why would that line be wrong?  We use `INTERRUPTSTACK < ALIGN_REQUIREMENT` to mean that there is no interrupt stack.  We are not changing `CONFIG_ARCH_INTERRUPTSTACK`.
But yes, when dumping the stack we always use the align down.  Not sure why, if we have a consistent INTSTACK_SIZE (whether aligned up or down) wouldn't that suffice?

Reply:
If the intension is aligned up, CONFIG_ARCH_INTERRUPTSTACK> 0 && CONFIG_ARCH_INTERRUPTSTACK < 7 is a correct stack value(even it's crazy that the stack has 8 bytes, but it correct from logic). Yes, as I mention before the key point is to use the alignup or aligndown consistently in all place, but since algindown is use in many place and there isn't real difference between alignup and aligndown, I would prefer to use the aligndown across the related code.

Reply:
@Ouss4, @xiaoxiang781216 
Thanks for the discussion.
Shall we keep this commit (i.e. remove +4) at line 326?

Reply:
If `CONFIG_ARCH_INTERRUPTSTACK> 0 && CONFIG_ARCH_INTERRUPTSTACK < 7` we won't align anything since we consider this value as "no interrupt stack".
But I understand and I agree with the rest of what you are saying.  We have to be consistent either aligning up or down. BTW, I decided to go with aligning up in my Xtensa PR (#2014)

> Shall we keep this commit (i.e. remove +4) at line 326?

Yes, I think we should keep the commit.

Reply:
@Ouss4 
Thanks, I keep the commit.

Reply:
Done

Reply:
Here is the thing. Nuttx does not support nested interrupt. If a design is super low on memory (it happens) the interrupt stack allocation has to not round down.  If  know the CONFIG (FPU etc) the size of the int stack is a fixed number . If I ask for That number and get a stack that is less it will over write memory.  This will be a debugging nightmare! Please round it UP

Reply:
This PR doesn't change the round behaviour, please don't do the different thing in one PR.

Reply:
need keep the old code after disscussion